### PR TITLE
Refactor logging - minimize Log.h

### PR DESCRIPTION
**Pull request by gumb0**

### CODE REVIEW

**Comment by gumb0** [comment]
@chfast clang6 and xcode90 are already at 8, I've increased gcc6 to 8 now

File: circle.yml [hunk 1]
```diff
@@ -138,7 +138,7 @@ jobs:
       - CC:  gcc-6
       - GCOV: gcov-6
       - GENERATOR: Ninja
-      - BUILD_PARALLEL_JOBS: 3
+      - BUILD_PARALLEL_JOBS: 8
```
Comment (chfast) [inline circle.yml]: Oh, sorry. I think it was 3 before LOG changes. You can increase it to 4 to check, but that's not required if it was not decreased by LOG changes.

Comment (gumb0) [reply circle.yml]: Ok (i think it was decreased in the beginning but then I rebased and kept the value from `develop`)

**Comment by gumb0** [comment]
Gcc6 still failed with 4 threads, I'll revert it

File: circle.yml (hunk 1 above)
Comment (chfast) [reply circle.yml]: GCC cannot handle 8 for sure, mostly because it also produces debug symbols.

Comment (gumb0) [reply circle.yml]: It's back to 3 now

**Pull request merged**